packages/smooth_app/lib/pages/scan/mkit_scan_helper.dart

```diff
@@ -0,0 +1,322 @@
+/// <- From the "decoder" Isolate
+///   -> When the Isolate is started (a [SendPort] is provided to communicate)
+///   -> When the Isolate is ready (camera description & scan mode are provided)
+///   -> When an image is decoded
+class _MLKitScanDecoderMainIsolate {
+  _MLKitScanDecoderMainIsolate({
+    required this.camera,
+    required this.scanMode,
+  }) : _port = ReceivePort() {
+    _port.listen((dynamic message) {
+      if (message is SendPort) {
+        _sendPort = message;
+
+        _sendPort!.send(
+          _MLKitScanDecoderIsolate._createConfig(
+            camera,
+            scanMode,
+          ),
+        );
+      } else if (message is bool) {
+        _isIsolateInitialized = true;
+
+        if (_queuedImage != null && _completer != null) {
+          _sendPort!.send(_queuedImage!.export());
+          _queuedImage = null;
+        }
+      } else if (message is List) {
+        _completer?.complete(message as List<String>);
+        _completer = null;
+      }
+    });
+  }
+
+  final CameraDescription camera;
+  final DevModeScanMode scanMode;
+
+  /// Port used by the Isolate to send us events:
+  /// - when a [SendPort] is available
+  /// - when the Isolate is ready to receive images
+  /// - when the Isolate has finished a detection
+  final ReceivePort _port;
+
+  /// Port provided by the Isolate to send instructions
+  /// - send the configuration (camera & scan mode)
+  /// - send a camera image
+  SendPort? _sendPort;
+
+  FlutterIsolate? _isolate;
+
+  /// Flag used when the Isolate is both started and ready to receive images
+  bool _isIsolateInitialized = false;
+
+  /// When an image is provided, this [Completer] allows to notify the response
+  Completer<List<String>?>? _completer;
+
+  /// When the Isolate is started, we have to wait until [_isIsolateInitialized]
+  /// is [true]. This variable temporary contains the image waiting to be
+  /// decoded.
+  CameraImage? _queuedImage;
+
+  /// Decodes barcodes from a [CameraImage]
+  /// A null result will be sent until the Isolate isn't ready
+  Future<List<String>?> decode(CameraImage image) async {
+    // If a decoding process is running -> ignore new requests
+    if (_completer != null) {
+      return null;
+    }
+
+    _completer = Completer<List<String>?>();
+
+    if (_isolate == null) {
+      _isolate = await FlutterIsolate.spawn(
+        _MLKitScanDecoderIsolate.startIsolate,
+        _port.sendPort,
+      );
+      _queuedImage = image;
+      return _completer!.future;
+    } else if (_isIsolateInitialized) {
+      _sendPort?.send(image.export());
+      return _completer!.future;
+    }
+
+    return null;
+  }
+
+  void dispose() {
+    _isIsolateInitialized = false;
+
+    _completer?.completeError(Exception('Isolate stopped'));
+    _completer = null;
+
+    _sendPort = null;
+
+    _isolate?.kill(priority: Isolate.immediate);
+    _isolate = null;
+  }
+}
+
+// ignore: avoid_classes_with_only_static_members
```

Comment:
I think adding a private empty constructor is better then doing it like that

Reply:
An isolate requires a top-level or static function, that's why :/